numba_dpcomp/numba_dpcomp/mlir_compiler/lib/pipelines/plier_to_scf.cpp

```diff
@@ -630,7 +636,7 @@ void PlierToScfPass::runOnOperation() {
   auto op = getOperation();
   (void)mlir::applyPatternsAndFoldGreedily(op, std::move(patterns));
 
-  op.walk([&](mlir::Operation *o) -> mlir::WalkResult {
+  op->walk([&](mlir::Operation *o) -> mlir::WalkResult {
```

Comment:
Is this the same as a check for legal ops?

Reply:
We cannot use legality check here as we don't  use dialect conversion framework.

Reply:
I am just trying to understand.
Shouldn't this use the conversion framework?

Reply:
This is actually much easier without conversion framework because you don't need to write so much legality checks